src/handler/batchPlateCreation.ts

```diff
@@ -0,0 +1,125 @@
+
+      logger.info(`Processing record: sysNum ${systemNumber}, timestamp ${createdTimestamp}`);
+
+      const dbRecord = await getBySystemNumberAndCreatedTimestamp(systemNumber, createdTimestamp);
+
+      if (!dbRecord || !Object.keys(dbRecord).length) {
+        throw new Error(`Missing record: sysNum ${systemNumber}, timestamp ${createdTimestamp}`);
+      }
+
+      validateRecord(dbRecord);
+
+      const newPlate: Plates = createNewPlate(batchIssuerName);
+
+      const formattedTechRecord = formatTechRecord<TechRecordTypeByVehicle<'hgv' | 'trl'>>(dbRecord);
+
+      if (isPlateAlreadyIssued(formattedTechRecord, batchIssuerName)) {
+        logger.info(`Plate already issued for: sysNum ${systemNumber}, timestamp ${createdTimestamp}`);
+        return;
+      }
+
+      updateRecordWithNewPlate(formattedTechRecord, newPlate);
+
+      const flattenedTechRecord = flattenArrays(formattedTechRecord) as TechRecordType<'get'>;
+      await inPlaceRecordUpdate(flattenedTechRecord);
+      numberOfRecordsUpdated++;
+
+      await sendToDocGenQueue(formattedTechRecord, newPlate);
+      numberOfSqsAdded++;
+
+      logger.info(`Successfully processed: sysNum ${systemNumber}, timestamp ${createdTimestamp}`);
+    });
+
+    await Promise.all(processPromises);
+
+    logger.info(`Batch Plate: Updated ${numberOfRecordsUpdated} tech records and added ${numberOfSqsAdded} to SQS`);
+  } catch (err: unknown) {
+    logError('Error in batch processing', err);
+    throw (err);
+  }
+};
+/**
+ * This function will validate the status code and the vehicle type on the technical record.
+ * @param dbRecord
+ */
+const validateRecord = (dbRecord: TechRecordType<'get'>) => {
+  if (dbRecord.techRecord_statusCode !== StatusCode.CURRENT) {
+    throw new Error(`Non current record: statusCode ${dbRecord.techRecord_statusCode}`);
+  }
+  if (dbRecord.techRecord_vehicleType !== 'trl' && dbRecord.techRecord_vehicleType !== 'hgv') {
+    throw new Error(`Invalid vehicle type: ${dbRecord.techRecord_vehicleType}`);
+  }
+};
+
+/**
+ * This function will create a new plate object.
+ * @param batchIssuerName
+ */
+const createNewPlate = (batchIssuerName: string): Plates => ({
+  plateSerialNumber: uuidv4(),
+  plateIssueDate: new Date().toISOString(),
+  plateReasonForIssue: PlateReasonForIssue.REPLACEMENT,
+  plateIssuer: batchIssuerName,
+});
+
+/**
+ * This function will check if a technical record already has a plate issued.
+ * @param record
+ * @param batchIssuerName
+ */
+const isPlateAlreadyIssued = <T extends TechRecordTypeByVehicle<'hgv' | 'trl'>>(
+  record: T,
+  batchIssuerName: string,
+): boolean => record.techRecord_plates?.some((plate) => plate.plateIssuer === batchIssuerName) ?? false;
+
+/**
+ * This function will update the technical record with the new plate that has been generated.
+ * @param record
+ * @param newPlate
+ */
+const updateRecordWithNewPlate = (record: TechRecordTypeByVehicle<'hgv' | 'trl'>, newPlate: Plates) => {
```

Comment:
do these need to be functions? are they going to be re-used?

Reply:
They're unlikely to be re used, just force of habbit to separate them! 😄 . Happy to put it in on big handler function though!

Reply:
Personally I think so. I think if things are complex pulling them out is very useful, but take for example the create plate, all we have done is basically move where the code is on the page.

I also think if they are going to be individual functions we should test them as such, so I would say just place them back in the handler :)